src/databricks/labs/ucx/runtime.py

```diff
@@ -203,6 +204,18 @@ def assessment_report(_: WorkspaceConfig):
     dashboard _before_ all tasks have been completed, but then only already completed information is shown."""
 
 
+@task("assessment", depends_on=[setup_schema])
+def workspace_listing(cfg: WorkspaceConfig, start_path: str = "/"):
+    """Scans the workspace for workspace objects. It recursively list all sub directories
+    and compiles a list of directories, notebooks, files, repos and libraries in the workspace.
+
+    Its uses multi-threading to parallelize the listing process to speed up on big workspaces.
+    It accepts starting path as the parameter defaulted to the root path '/'."""
+    ws = WorkspaceClient(config=cfg.to_databricks_config())
+    ws_listing = WorkspaceListing(ws, num_threads=cfg.num_threads)
+    ws_listing.walk(start_path)
```

Comment:
and how would `crawl_permissions` know about the results of it? make it extend the `CrawlerBase` :)

Reply:
have done the changes. please check. I have not touched the 
https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/workspace_access/generic.py
WorkspaceListing class which is doing the workspace listing. Should we change that to fetch from the new crawler?
@nfx